src/lumigo_tracer/wrappers/http/sync_http_wrappers.py

```diff
@@ -245,10 +246,15 @@ def _headers_reminder_wrapper(func, instance, args, kwargs):
     Remember the headers helps us to improve performances on requests that use this flow.
     """
     with lumigo_safe_execute("add hooked data"):
+        headers = safe_get_list(args, HEADERS_ARG_INDEX_REQUEST) or kwargs.get("headers") or {}
+        if Configuration.propagate_w3c:
+            SpansContainer.get_span().add_w3c_trace_propagator(headers)
+            kwargs["headers"] = headers
```

Comment:
The function `add_w3c_trace_propagator` already modifies the `headers` object so I think that this line is redundant.

Reply:
Nope! Look at the previous comment :)
(This is the case where no headers were given!